src/translators/rust/pure/cargo-lock/default.nix

```diff
@@ -171,6 +172,13 @@ in {
       # The structure of this should be defined in:
       #   ./src/specifications/{subsystem}
       subsystemAttrs = rec {
+        replacePathsWithAbsolute = l.listToAttrs (l.map (
+            project: {
+              name = project.relPath;
+              value = "${rootSource}/${project.relPath}";
+            }
+          )
+          discoveredProjects);
```

Comment:
I guess you might want to filter for project with subsystem `rust` only.
Also I think this will not always work. discoveredProjects only contains projects with a Cargo.lock file. So whenever a project without Cargo.lock is referenced it will be missing here.

Why don't we move some of that logic to the discoverer, so the discovered projects will look like that:
```json
[
  {
    "name": "foo",
    "subsystem": "rust",
    "subsystemInfo": {
      "referredWorkspaces": [
        "rel/path/to/arbitrary/directory/within/source"
      ],
    },
  }
]
```

With this, you will get `referredWorkspaces` as an input to the translator, and you will have all you need, right?

Reply:
Yeah, I was doing that actually, but didn't push that yet.

Something i noticed with this approach is that for patching the paths we need to create the proper relative path. So for example, in `Cargo.toml`, a path dependency will have the path `../../library`. So we need to be able to create the `../../library` from our list of crates, which means we need to figure out the relative paths to the project we are translating. Then we put those in the `subsystemAttrs`, relative path as the key, and the absolute path as the value, then we just run substitute on those. Do you happen to have any idea on how resolve those relative paths efficiently?

Reply:
I just added a library function `sanitizePath` via b68a9ee61c733aa8aec6059f439e184eb6cf85df.
Now you can do this:
```command
# from the dream2nix source
nix-repl> :lf ./.
Added 14 variables.

nix-repl> tree = lib.dlib.prepareSourceTree {source = ./.;}

nix-repl> subTree = tree.getNodeFromPath "src/fetchers"

nix-repl> lib.dlib.sanitizePath "${subTree.fullPath}/../"
"/nix/store/26k11das6x5n4449sywhdkqrjw3nfzif-dream2nix/src"
```

Does that solve the problem?
There is also `sanitizeRelativePath` to do the same for relative paths.

Reply:
Yep! I was able to use that to fix how I do things. It seems to be working now. I have updated https://github.com/yusdacra/dream2nix-issue-113, so you can check if it works for you too there.